### ISSUE TITLE
Allow us to rename the addons folder to Addons

### ISSUE THREAD

**Issue by Shadowblitz16** [godotengine/godot-proposals#10536]
### Describe the project you are working on

Nothing right now. just experimenting with 4.3

### Describe the problem or limitation you are having in your project

I am using C# but my plugins don't show up because I put them in a PascalCase folder called Addons so they would be consistent with my other folder names trying to be consistent with C# script naming limitations

If you don't know what I mean by C# script naming limitations godot at least last time I tested it requires the file name to be the same as the c# class name

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Allow us to rename addons folder to Addons

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

it would either be dynamic in which users can rename thier addons folder to anything or just a additional check for a folder named `Addons`.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

It would be used with C# devs and people with ocd

### Is there a reason why this should be core and not an add-on in the asset library?

The plugin manager already allows us to place addons in a different folder but they don't show up on the addons list in project settings.

**Comment by tokengamedev** [comment]
I thought this was working. 
Only time it did not work when I changed to a different folder name like `my_addons`.

the only thing I see to is to map the addons folder into an editor settings. That way the name will not be hardcoded and by default addons folder can still be utilized.
Same thing applies for export folders like android

**Comment by Shadowblitz16** [comment]
> I thought this was working.

Not for GDExtensions as far as I can tell.

**Comment by Calinou** [comment]
One reason why we might want want to use `Addons` in C# projects is that not all C# projects will *only* use add-ons written in C#. The add-ons written in GDScript will remain in a lowercase `addons` folder, as it's the path they are originally expecting (if you downloaded existing add-ons that contain absolute paths).

**Comment by Shadowblitz16** [comment]
> One reason why we might **not** want to use `Addons` in C# projects is that not all C# projects will _only_ use add-ons written in C#. The add-ons written in GDScript will remain in a lowercase `addons` folder, as it's the path they are originally expecting (if you downloaded existing add-ons that contain absolute paths).
> 
> Having `addons` and `Addons` in the same folder is not technically possible on case-insensitive filesystems (i.e. on Windows and recent macOS versions).

I am not talking about two different folders per script type. 
I am talking about consistency, one or the either, it would probably be a project settings thing where you can change the folder name (case sensitive)

**Comment by Calinou** [comment]
> I am talking about consistency, one or the either, it would probably be a project settings thing where you can change the folder name (case sensitive)

The thing is that existing add-ons expect the add-ons folder path to be lowercase `addons`. They would not be aware of the project setting that certain projects might change, so these add-ons would suddenly appear to break when installed into certain projects (but not others that kept the default value).

**Comment by Shadowblitz16** [comment]
> > I am talking about consistency, one or the either, it would probably be a project settings thing where you can change the folder name (case sensitive)
> 
> The thing is that existing add-ons expect the add-ons folder path to be lowercase `addons`. They would not be aware of the project setting that certain projects might change, so these add-ons would suddenly appear to break when installed into certain projects (but not others that kept the default value).

Maybe a compatibility layer could be made?
How exactly do existing addons get the addon folder path? just hardcoded in the plugin code?

**Comment by Calinou** [comment]
> How exactly do existing addons get the addon folder path? just hardcoded in the plugin code?

Yes, it's hardcoded in `preload()` (which must be a constant expression) and `load()`.
